### PR TITLE
Groups cluster: XML types fixed.

### LINKED ISSUE #10335: Fix command arguments for Get Group Membership command
#### Problem
There should just be a list, no out-of-band length.

 #### Proposed Solution
Fix XML, fix implementation.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
Groups cluster XML uses UINT16 for group IDs. The correct type is `group_id`
* Fixes [#10335: Command arguments for Get Group Membership command](https://github.com/project-chip/connectedhomeip/issues/10335).

#### Change overview
UINT16 replaced with group_id for all group IDs within the groups cluster XML.

#### Testing
* Project built successfully.
* YAML tests ran successfully.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

**Review by jmeg-sfy — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/silabs/general.xml [hunk 1]
```diff
@@ -219,37 +219,37 @@ limitations under the License.
...
       <arg name="status" type="ENUM8"/>
-      <arg name="groupId" type="INT16U"/>
+      <arg name="groupId" type="group_id"/>
       <arg name="groupName" type="CHAR_STRING"/>
     </command>
     <command source="server" code="0x02" name="GetGroupMembershipResponse" optional="false" disableDefaultResponse="true">
       <description>
         Command description for GetGroupMembershipResponse
       </description>
       <arg name="capacity" type="INT8U"/>
```
[the remark this answers is not in the document]
Comment (rcasallas-silabs) [reply src/app/zap-templates/zcl/data-model/silabs/general.xml]: Capacity flagged as nullable.

File: src/app/clusters/groups-server/groups-server.cpp [hunk 2]
```diff
@@ -197,7 +197,7 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
 struct GroupMembershipResponse
 {
     // A capacity of 0xFF means that it is unknown if any further groups MAY be added.
-    static constexpr uint8_t kCapacityUnknown = 0xff;
+    const chip::app::DataModel::Nullable<uint8_t> kCapacityUnknown = chip::app::DataModel::Nullable<uint8_t>(0xff);
```
Comment (bzbarsky-apple) [inline src/app/clusters/groups-server/groups-server.cpp:200]: This is wrong.  Per spec unknown means null, not 0xFF.

In fact, as far as I can tell encoding this (`Nullable<uint8_t>(0xff)`) should be failing.  Are we not actually testing this case?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/groups-server/groups-server.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/clusters/groups-server/groups-server.cpp:200]: Ah, we are in fact testing it, looks like, and those tests are failing....

Comment (bzbarsky-apple) [inline src/app/clusters/groups-server/groups-server.cpp:200]:
```suggestion
    const chip::app::DataModel::Nullable<uint8_t> kCapacityUnknown;
```

File: src/app/clusters/groups-server/groups-server.cpp [hunk 3]
```diff
@@ -197,7 +197,7 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
 struct GroupMembershipResponse
 {
     // A capacity of 0xFF means that it is unknown if any further groups MAY be added.
```
Comment (bzbarsky-apple) [inline src/app/clusters/groups-server/groups-server.cpp:199]:
```suggestion
    // A null capacity means that it is unknown if any further groups MAY be added.
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/groups-server/groups-server.cpp (hunk 2 above)
Comment (rcasallas-silabs) [reply src/app/clusters/groups-server/groups-server.cpp]: Unknown used to be 0xff, but it is now NULL. Fixed.

Comment (rcasallas-silabs) [reply src/app/clusters/groups-server/groups-server.cpp]: Fixed as suggested.

File: src/app/clusters/groups-server/groups-server.cpp (hunk 3 above)
Comment (rcasallas-silabs) [reply src/app/clusters/groups-server/groups-server.cpp]: Fixed as suggested.

**Comment by bzbarsky-apple** [comment]
"Test Failure: capacity expected to not be null but is" is what the YAML test failure is.  Probably needs updates to the test expectations?

**Pull request merged**